sound/soc/sof/sof-client-probes.c

```diff
@@ -188,6 +194,56 @@ static const struct snd_compress_ops sof_probes_compressed_ops = {
 	.copy = sof_probes_compr_copy,
 };
 
+#ifdef CONFIG_SND_SOC_SOF_IPC4
+static const char *sof_probe_ipc4_type(u32 type)
+{
+	switch (type) {
+	case PROBE_TYPE_INPUT:
+		return "input";
+	case PROBE_TYPE_OUTPUT:
+		return "output";
+	case PROBE_TYPE_INTERNAL:
+		return "internal";
+	default:
+		return "UNKNOWN";
+	}
+}
+
+static int sof_probes_ipc4_prints(struct sof_client_dev *cdev, char *buf, size_t size,
+				  struct sof_probe_point_desc *desc)
+{
+	struct snd_sof_dev *sdev = sof_client_dev_to_sof_dev(cdev);
```

Comment:
I'm rather uneasy with sof-client modules using sdev, it is only allowed in sof-client core module.